rm: security: update stm32mp1 article (#588)
foundriesio/docs: source/reference-manual/security/secure-boot-stm32mp1.rst

directory and put all these image along with unsigned binaries::

        flashlayouts-stm32mp1$ ls -lah
        total 558M
        drwxr-xr-x 2 user user 4,0K aug 17 12:47 .
        drwxr-xr-x 9 user user  12K aug 17 12:48 ..
        -rwxr-xr-x 1 user user  28K aug 17 12:47 create_sdcard_from_flashlayout.sh
        -rw-r--r-- 1 user user 1,5M aug 17 12:47 fip-stm32mp157c-dk2-optee.bin
        -rwxr-xr-x 1 user user  609 aug 17 12:47 FlashLayout_sd_stm32mp157c-dk2-optee.tsv
        -rw-r--r-- 1 user user 709M aug 17 12:47 lmp-base-console-image-stm32mp15-disco-sec.ext4
        -rw-r--r-- 1 user user 198K aug 17 12:47 tf-a-stm32mp157c-dk2-sdcard_Signed.stm32
        -rw-r--r-- 1 user user 198K aug 17 12:47 tf-a-stm32mp157c-dk2-sdcard.stm32
        -rw-r--r-- 1 user user 194K aug 17 12:47 tf-a-stm32mp157c-dk2-uart_Signed.stm32
        -rw-r--r-- 1 user user 194K aug 17 12:47 tf-a-stm32mp157c-dk2-uart.stm32
        -rw-r--r-- 1 user user 198K aug 17 12:47 tf-a-stm32mp157c-dk2-usb_Signed.stm32
        -rw-r--r-- 1 user user 198K aug 17 12:47 tf-a-stm32mp157c-dk2-usb.stm32

Provision PKH, HUK and RPMB
^^^^^^^^^^^^^^^^^^^^^^^^^^^

> `stm32-mfgtool-files` recipe contains all needed tools for provisioning
> PKH/HUK and programming RPMB key on the destination device.
> To use it, build it using `lmp-mfgtool` distro. For example::

Reviewer: ```suggestion
The ``stm32-mfgtool-files`` recipe contains the tools needed for provisioning
PKH/HUK, and programming the RPMB key on the destination device.
To use, build the ``lmp-mfgtool`` distro. For example::
```

        $ DISTRO=lmp-mfgtool MACHINE=stm32mp15-eval-sec . setup-environment
        $ bitbake stm32-mfgtool-files

.. warning::

   If automatic signing is disabled, you need to sign all boot images manually
   and put signed binaries to `stm32-mfgtool-files` (similar to `flashlayouts-stm32mp1`)
   directory before executing `provision.sh` script. For details check
   `Sign and Deploy the BL2 Image Manually` section.

> Switch to `root` user and add the path to STM32Cube to your PATH env variable::
> 
>         $ sudo -s
>         # export PATH=$PATH:<path_to_stm32cube>

Reviewer: ```suggestion
Switch to ``root`` user, and add the path to STM32Cube to ``PATH``:

  .. prompt:: bash $,#, auto
      
        $ sudo -s
        # export PATH=$PATH:<path_to_stm32cube>
```

> Execute script, providing path to the PKH binary file::

Reviewer: ```suggestion
Execute script, providing path to the PKH binary file:

 .. prompt:: bash #, auto
```

should be fused to OTP and  used to authenticate the public key on the
target. For more details refer to ST's `STM32 KeyGen tool`_ guide.

How to Secure the Platform (Automatic Approach)
-----------------------------------------------

This approach can be used for both STM32MP15 Discovery and Evaluation kits.

> Automatic Signing Using Yocto

Reviewer 2: ```suggestion
Automatic Signing Using the Yocto Project
```
But I think it's weird to have yocto mentioned here. Maybe it's LmP? Anyway, if we mention yocto, we have to use the complete name

Author: changed to LmP

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> The FSBL binary (TF-A BL2) must be signed using a generated key pair.

Author: I've extended the sentence adding info about what key pair should be used

There are two possible ways to do this: automatic implicit signing during
build time, and manually. In both cases, `STM32 Signing tool`_ needs to
be installed.

To enable implicit automatic signing of boot images during build time
uncomment these lines in your `conf/local.conf`::

        #
        # STM32CubeProgrammer STM32MP Signing Tool configuration
        #
        STM32_ROT_SIGN_ENABLE ??= "1"
        STM32_CUBE_PATH ??= "/usr/local/STMicroelectronics/STM32Cube/STM32CubeProgrammer"
        STM32_ROT_KEY_PATH ??= "${TOPDIR}/../tools/lmp-tools/security/stm32mp1/"
        STM32_ROT_KEY_PATH[vardepsexclude] += "TOPDIR"

>         STM32_ROT_KEY_PASSWORD ??= "foundries"

Reviewer 2: Could it be created a factory variable to enable this from the factory-config.yml?

Author: I really doubt that it's good idea to share RoT keys on CI server.
I would use this approach only on for building final boot image on private host.

        total 558M
        drwxr-xr-x 2 user user 4,0K aug 17 12:47 .
        drwxr-xr-x 9 user user  12K aug 17 12:48 ..
        -rwxr-xr-x 1 user user  28K aug 17 12:47 create_sdcard_from_flashlayout.sh
        -rw-r--r-- 1 user user 1,5M aug 17 12:47 fip-stm32mp157c-dk2-optee.bin
        -rwxr-xr-x 1 user user  609 aug 17 12:47 FlashLayout_sd_stm32mp157c-dk2-optee.tsv
        -rw-r--r-- 1 user user 709M aug 17 12:47 lmp-base-console-image-stm32mp15-disco-sec.ext4
        -rw-r--r-- 1 user user 198K aug 17 12:47 tf-a-stm32mp157c-dk2-sdcard_Signed.stm32
        -rw-r--r-- 1 user user 198K aug 17 12:47 tf-a-stm32mp157c-dk2-sdcard.stm32
        -rw-r--r-- 1 user user 198K aug 17 12:47 tf-a-stm32mp157c-dk2-emmc_Signed.stm32
        -rw-r--r-- 1 user user 198K aug 17 12:47 tf-a-stm32mp157c-dk2-emmc.stm32
        -rw-r--r-- 1 user user 194K aug 17 12:47 tf-a-stm32mp157c-dk2-uart_Signed.stm32
        -rw-r--r-- 1 user user 194K aug 17 12:47 tf-a-stm32mp157c-dk2-uart.stm32
        -rw-r--r-- 1 user user 198K aug 17 12:47 tf-a-stm32mp157c-dk2-usb_Signed.stm32
        -rw-r--r-- 1 user user 198K aug 17 12:47 tf-a-stm32mp157c-dk2-usb.stm32

Create ``combo*`` images using signed binaries::

>         flashlayouts-stm32mp1$ cp tf-a-stm32mp157c-dk2-emmc_Signed.stm32 combo-emmc-tfa-fip-stm32mp157c-ev1.bin

Reviewer 2: is this executed on the host or on the target?

Author: host, I've extented description

        HASK key is not locked!

> Sign and Deploy the BL2 Image Manually

Reviewer: ```suggestion
.. _deploy-bl2-manually:

Sign and Deploy the BL2 Image Manually
```

                       01 df ae 4c cd 99 12 bc d3 fc 9b 30 7a 77 c5 2b f0 5b 01 f3 2e bb c3 71 db a4 40 93 2c 01 3f a2
        Binary type: 0x10

> In case WIC image is used for flashing, you have to deploy manually all signed images to the SD card after flashing WIC image
> to SD, where existing non signed images must be replaced. That can be achieved with a simple ``dd`` command as well

Reviewer: ```suggestion
In the case where the  WIC image is used for flashing, you have to manually deploy the signed images to the SD card after flashing the WIC image.
Here the existing non signed images must be replaced. This can be achieved with a simple ``dd`` command as well
```
